### ISSUE TITLE
Support XEP-0280 Message Carbons

### ISSUE THREAD

**Issue by aredridel** [jabberd2/jabberd2#55]
Totally needed to make XMPP sane with intermittent connectivity and device switching

**Comment by texpert** [comment]
+1

**Comment by smokku** [comment]
Jabberd 2 supports many resources with the same priority. If there are many resources with highest priority, the message is sent to all resources.
This is a standard, RFC defined behavior.

Why do we need an extension to reimplement it in non-RFC way?

**Comment by crosser** [comment]
xep-0280 allows you to have complete consistent chat history on all clients. I.e. not only your peer's, but also your own sent messages show up in the chat history on all your devices.

**Comment by smokku** [comment]
OK. While I do see the use-case, it really feels like a band-aid.
This does not allow for "chat history" - messages are not persisted. Also "all clients" is an overstatement - only _connected_ clients benefit from carbons. If client connection gets interrupted, it will loose a part of the conversation, no matter whether it has carbons enabled. :-/

Carbons together with XEP-0136 automatic archiving would support the use case of a shared conversation on all devices. Yet I have a feeling these should be one XEP.

**Comment by crosser** [comment]
Indeed. The way Hangouts (and others?) do it, when you have fully synchronized chat history on any device, is more intuitive.
Perhaps chat service should have been designed not as message delivery, but as syncing of an append-only log, in the first place...

**Comment by smokku** [comment]
This is what https://en.wikipedia.org/wiki/Matrix_(communication_protocol)#Protocol does. Makes sense for distributed chat.
On the other hand XMPP is much more than people talking people and simple, reliable, throw-away messages is what M2M needs and XMPP excels at.

**Comment by lameventanas** [comment]
How about implementing both XEP-0280 and XEP-0313: Message Archive Management?

I think that would solve all the problems, wouldn't it?
